kaska/s2_observations.py

```diff
@@ -219,133 +263,181 @@ def read_time_series(self, time_grid):
-
+        rho_surface[:, ~mask] = np.nan
+        # uncertainty
         rho_unc = np.array(rho_unc) / 10000.0
         rho_unc[:, ~mask] = np.nan
         # Average uncertainty over the image
         rho_unc = np.nanmean(rho_unc, axis=(1, 2))
-        # Set missing pixels to NaN
-        rho_surface[:, ~mask] = np.nan
-        # Now read angles
+
+        # Read in angles
+        #print(next(current_folder.glob("**/SAA_SZA.tif")))
+        try:
+            angle_file_S = next(current_folder.glob("**/SAA_SZA.tif"))
+            #print(angle_file_S)
+        except StopIteration:
+            LOG.info(f"Sun angle file SAA_SZA.tif does not exist.")
+            return None, None, None, None, None, None
+        try:
+            angle_file_V = next(current_folder.glob("**/VAA_VZA_B05.tif"))
+        except StopIteration:
+            LOG.info(f"View angle file VAA_VZA_B05.tif does not exist.")
+            return None, None, None, None, None, None
+
         sun_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/SAA_SZA.tif"),
+            str(angle_file_S),
             target_img=self.state_mask,
             x_res=20,
             y_res=20,
             resample=0,
         ).ReadAsArray()
         view_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/VAA_VZA_B05.tif"),
+            str(angle_file_V),
             target_img=self.state_mask,
             x_res=20,
             y_res=20,
             resample=0,
         ).ReadAsArray()
+        # zenith angles: 90 - Altitude
         sza = np.cos(np.deg2rad(sun_angles[1].mean() / 100.0))
         vza = np.cos(np.deg2rad(view_angles[1].mean() / 100.0))
+        # azimuth
         saa = sun_angles[0].mean() / 100.0
         vaa = view_angles[0].mean() / 100.0
         raa = np.cos(np.deg2rad(vaa - saa))
         return rho_surface, mask, sza, vza, raa, rho_unc
 
+    def _read_band_data(self, what, band, s2_files):
+        """
+        Reads different types of data for the given band from the list of
+        input files.
+
+        Parameters
+        ----------
+        what : the type of data to read. Currently 'rho' for reflectivity or
+               'unc' for uncertainty
+        band : the name of the band (str)
+        s2_files : a list of files (paths) containing the data for all bands
+                   of one timestep.
+        Returns
+        -------
+        numpy array containing the (reflectivity or uncertainty) data for
+        one band.
+        """
+
+        meta_dict = {'rho' : ('', ''),
+                     'unc' : ('_unc', 'uncertainty')}
+
+        try:
+            original_s2_file = next(
+                f for f in s2_files
+                if str(f).endswith(f"{band}_sur{meta_dict[what][0]}.tif")
+            )
+        except KeyError:
+            LOG.info(f"""Only reflectivity ('rho') or uncertainty ('unc') data
+             read by _read_band_data. Calling argument {what} not recognised.
+            """)
```

Comment:
strings as `"""` will keep the spaces of the middle line. Though I'm not sure how that will work within a log.

Reply:
I changed those to f-strings concatenations.